Tarea clase6 ejercicio2

guido732 opened the pull request:
![imagen](https://user-images.githubusercontent.com/23202677/66777299-0efa3480-ee9f-11e9-9695-ab74d636a3f5.png)

guido732 committed:
Agrega elementos HTML y estilos CSS base

guido732 committed:
Agrega funcionalidad "Agregar Elemento" al script

refactor de código del ejercicio 1 para funcionalidad de agregar elemento

guido732 committed:
Agrega funcionalidad de eliminar elemento

Agrega funcionalidad al botón eliminar familiar para remover elementos generados, en caso que no haya elementos para prevenir errores se hace un return false

guido732 committed:
Agrega estilos para prevenir desplazamiento del formulario

Fija el formulario al tope de la pantalla para prevenir desplazamiento al agregar elementos nuevos o eliminar, mejorando la experiencia de usuario

guido732 committed:
Agrega funcionalidad cálculos y corrige nombres variables

guido732 committed:
Agrega estilos para cuando hay overflow de elementos en el eje Y

guido732 committed:
Arregla elementos de posicion del output y estilos

guido732 committed:
Agrega funcionalidad de ignorar si el valor es 0

guido732 committed:
Corrige validación de inputs para valores negativos

Agrega función limpiar array para calculo de sueldo mínimo solo con valores positivos dentro del array y error prevention en caso que el array procesado sea vacío (undefined) se tome como un valor inválido y devuelva error pidiendo número váldio

guido732 committed:
Agrega Estilos a los botones y corrige multiples valores de output al presionar calcular

guido732 committed:
Ordena el script (btns press y despues funciones)

guido732 committed:
Agrega redondeo de decimales a 0.00

guido732 committed:
Agrega sueldo mensual promedio

fsodano commented on tareas/tarea-6/ejercicio2/main.js:
```diff
@@ -0,0 +1,151 @@
+	const paragraphs = document.querySelectorAll(".paragraph-element");
+	paragraphs.forEach(element => {
+		element.remove();
+	});
```
Si anidaras todos los elementos dentro de un único elemento con clase `grupo`, por ejemplo, podrías sólo remover ese `grupo` y hacer un único for.

guido732 replied:
Corregido

fsodano commented on tareas/tarea-6/ejercicio2/main.js:
```diff
@@ -0,0 +1,151 @@
+Punto bonus: si hay inputs vacíos, ignorarlos en el cálculo (no contarlos como 0).
+*/
+
+let inputCounter = 0;
```
Intentá no depender de variables globales. Este `inputCounter` hace lo mismo que `document.querySelectorAll('input').length`, ponele

guido732 replied:
Corregido

fsodano commented on tareas/tarea-6/ejercicio2/main.js:
```diff
@@ -0,0 +1,151 @@
+		return "Ingresá al menos un sueldo válido";
+	} else {
+		return sueldoMinimo.toFixed(2);
+	}
```
Si bien no lo vimos y esta solución es aceptable, si te interesa, leé acerca de tirar excepciones. Es la forma adecuada de reportar que hay un error, en vez de hacer que esta función devuelva 2 tipos de datos (string y number).

guido732 replied:
Ahí estuve leyendo un poco. 
Reemplacé el return por throw en ese caso y me lo devuelve por consola, sin embargo el programa funciona como antes (sigue agregando ese valor al elemento <p> generado. Ésto es correcto?

guido732 replied:
https://github.com/guido732/introduccion-a-js/pull/6/commits/0dad48b904bfb43b9a8216380b1d216e8c8e07e9

Ese es el commit por cualquier cosa

fsodano commented on tareas/tarea-6/ejercicio2/main.js:
```diff
@@ -0,0 +1,151 @@
+}
+function crearElementoParrafo(valorInterno, nombreFuncion) {
+	const nuevoParrafo = document.createElement("p");
+	nuevoParrafo.classList.add("paragraph-element");
```
Este nombre debería ser más semántico, por ejemplo: `stat` ó `info`

guido732 replied:
Corregido

guido732 committed:
Agrega clase generalizada a elementos generados y remueve código innecesario

Agrega .generated-element como clase a todos los elementos creados por el usuario para simplificar la eliminación de los mismos

guido732 committed:
Remplaza una variable global por un queryselector a todos los inputs

Para evitar el uso de una variable global innecesaria que contaba la cantidad de inputs se reemplazó la misma por un queryselector que cuenta todos los inputs existentes simplificando el código

guido732 committed:
Agrega error handling

Reemplaza return por throw para hacer

guido732 committed:
Modifica nombre de clase para elemento <p> generado

Modifica el nombre de la clase para hacer el mismo más semántico y mejorar la lectura del código
